### PR TITLE
[#1239] Add file type label in commits panel

### LINKED ISSUE #1239: Commits Panel: add filetype tag to each commit
With the recent merged #1192, we have the file type info of each commit. 

With this info, we can easily add the filetype tag(s) to each commit and allowing the filter of commits based on filetypes.  Similar to the UI implemented [here](https://github.com/reposense/RepoSense/pull/1093), but in Commits Panel.

### PR DESCRIPTION
**Pull request by jamessspanggg**
Fixes #1239 

```
With the recent merged #1192, we have the file type info of each commit.

With this info, we can easily add the filetype label(s) to each commit.

Let's add the filetype label(s) beside each commit.
```

### CODE REVIEW

**Comment by jamessspanggg** [comment]
On hold until #1093 merges

**Comment by jamessspanggg** [comment]
@damithc may I get ur initial review on the UI? Currently I just have a `show file type in commits` checkbox which toggles the visibility of the file type in commits.

**Comment by Tejas2805** [comment]
Looking at this PR, I feel we should provide some kind of box for the commit titles too. Will loo neater.

**Comment by 0blivious** [comment]
I feel the UI for show all button can be improved and maybe better to standardise the size of tag and file type label. Also is it possible to maintain the sequence of the labels as the one shown in code view?

File: frontend/src/static/css/panels.scss [hunk 1]
```diff
@@ -87,6 +87,14 @@
   }
 }
 
+.fileTypeLabel {
```
Comment (0blivious) [inline frontend/src/static/css/panels.scss:90]: I remembered I have already shifted this to panels in the previous PR?

File: frontend/src/static/js/v_summary.js [hunk 2]
```diff
@@ -857,6 +857,7 @@ window.vSummary = {
     this.$root.$on('restoreCommits', (info) => {
       const zoomFilteredUser = this.restoreZoomFiltered(info);
       info.zUser = zoomFilteredUser;
+      info.zFileTypeColors = this.fileTypeColors;
```
Comment (0blivious) [inline frontend/src/static/js/v_summary.js:860]: Just leave a mark here, maybe should shift fileTypeColors to global constant after vuex implemented

File: frontend/src/static/css/panels.scss (hunk 1 above)
Comment (0blivious) [reply frontend/src/static/css/panels.scss:90]: And is it possible to merge this with fileTypes css style?

**Comment by jamessspanggg** [comment]
Won't be having enough time to continue with this PR. Anyone can feel free to take over this PR.

**Pull request closed without merging**